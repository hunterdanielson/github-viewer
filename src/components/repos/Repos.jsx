import React from 'react';
import PropTypes from 'prop-types';
// import { getPRs } from '../../services/gitHubApi';

const Repos = ({ repos }) => (
  <>
    <h2>Repos</h2>
    <ul>
      {
        repos.map(repo => <li key={repo.id}>
          {repo.name}
          <ul>
            {/* {getPRs(gitHubProfile, repo.name)
              .map(pull => <li key={pull.id}>{pull.number}</li>)} */}
          </ul>
        </li>)
      }
    </ul>
  </>
);

Repos.propTypes = {
  // gitHubProfile: PropTypes.string.isRequired,
  repos: PropTypes.array
};

export default Repos;
