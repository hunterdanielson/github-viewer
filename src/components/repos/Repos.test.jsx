import React from 'react';
import { shallow } from 'enzyme';
import Repos from './Repos';

describe('Repos component', () => {
  it('renders Repos', () => {
    const wrapper = shallow(<Repos repos={[{ name: 'test', id: 1 }, { name: 'tester', id: 2 }]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
