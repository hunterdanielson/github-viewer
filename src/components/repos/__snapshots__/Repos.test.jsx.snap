// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos component renders Repos 1`] = `
<Fragment>
  <h2>
    Repos
  </h2>
  <ul>
    <li
      key="1"
    >
      test
      <ul />
    </li>
    <li
      key="2"
    >
      tester
      <ul />
    </li>
  </ul>
</Fragment>
`;
